planner/mapper/mapper.go

```diff
@@ -119,36 +120,91 @@ func resolveOrderDependencies(
 
 	// If there is orderby, and any one of the condition fields that are join fields and have not been
 	// requested, we need to map them here.
+outer:
 	for _, condition := range source.Value().Conditions {
-		if len(condition.Fields) <= 1 {
-			continue
-		}
-
-		joinField := condition.Fields[0]
+		fields := condition.Fields[:] // copy slice
+		for {
+			numFields := len(fields)
+			if numFields == 2 {
+				joinField := condition.Fields[0]
+
+				// ensure the child select is resolved for this order join
+				innserSelect, err := resolveChildOrder(descriptionsRepo, descName, joinField, mapping, existingFields)
+				if err != nil {
+					return err
+				}
 
-		// Check if the join field is already mapped, if not then map it.
-		if isOrderJoinFieldMapped := len(mapping.IndexesByName[joinField]) != 0; !isOrderJoinFieldMapped {
-			index := mapping.GetNextIndex()
-			mapping.Add(index, joinField)
+				// make sure the actual target field inside the join field
+				// is included in the select
+				targetFieldName := condition.Fields[1]
+				targetField := &Field{
+					Index: innserSelect.FirstIndexOfName(targetFieldName),
+					Name:  targetFieldName,
+				}
+				innserSelect.Fields = append(innserSelect.Fields, targetField)
+				continue outer
```

Comment:
todo: The two `continue outer` statements appear to be redundant, as there is nothing else within the loop that would otherwise be executed.  Please remove them along with the declaration of `outer`.